ReminderApp-OBJC/ReminderApp-OBJC/Classes/Controller/CDReminderTableViewController.m

```diff
@@ -129,62 +101,231 @@ - (void)controller:(NSFetchedResultsController *)controller didChangeObject:(id)
 	  newIndexPath:(NSIndexPath *)newIndexPath {
 	
 	switch (type) {
-		case NSFetchedResultsChangeInsert:
-			[self.tableView insertRowsAtIndexPaths:[NSArray arrayWithObject:newIndexPath]
+  		case NSFetchedResultsChangeInsert:
+			[self.tableView insertRowsAtIndexPaths:@[newIndexPath]
 								  withRowAnimation:UITableViewRowAnimationFade];
 			break;
 		case NSFetchedResultsChangeDelete:
-			[self.tableView deleteRowsAtIndexPaths:[NSArray arrayWithObject:indexPath]
+			[self.tableView deleteRowsAtIndexPaths:@[indexPath]
 								  withRowAnimation:UITableViewRowAnimationFade];
 			break;
 		case NSFetchedResultsChangeUpdate: {
-			UITableViewCell *cell = [self.tableView cellForRowAtIndexPath:indexPath];
+			CDAddReminderCell *cell = [self.tableView cellForRowAtIndexPath:indexPath];
 			CDReminder *reminderItem = [self.fetchedResultsController objectAtIndexPath:indexPath];
 			cell.textLabel.text = reminderItem.taskName;
-			cell.detailTextLabel.text = reminderItem.taskHoure;
 		}
 			break;
 		case NSFetchedResultsChangeMove:
-			[self.tableView deleteRowsAtIndexPaths:[NSArray arrayWithObject:indexPath]
+			[self.tableView deleteRowsAtIndexPaths:@[indexPath]
 								  withRowAnimation:UITableViewRowAnimationFade];
-			[self.tableView insertRowsAtIndexPaths:[NSArray arrayWithObject:newIndexPath]
+			[self.tableView insertRowsAtIndexPaths:@[newIndexPath]
 								  withRowAnimation:UITableViewRowAnimationFade];
 			break;
-		default:
-			break;
 	}
 	
 }
 
-- (void) controllerDidChangeContent:(NSFetchedResultsController *)controller {
+- (void)controllerDidChangeContent:(NSFetchedResultsController *)controller {
 	
 	[self.tableView endUpdates];
 	
 }
 
-#pragma mark - DataSource and Delegate
+#pragma mark - Helper methods
+
+- (CGFloat)textViewHeightForRowAtIndexPath:(NSIndexPath *)indexPath {
 
-- (NSInteger)numberOfSectionsInTableView:(UITableView *) tableView {
+	UITextView *calculationView = ((CDAddReminderCell *)[self.tableView cellForRowAtIndexPath:indexPath]).textView;
+	CGFloat textViewWidth = calculationView.frame.size.width;
 	
-	return self.fetchedResultsController.sections.count;
+	if (!calculationView) {
+		return 50;
+	}
 
+	CGSize size = [calculationView sizeThatFits:CGSizeMake(textViewWidth, FLT_MAX)];
+	return size.height;
+	
 }
 
-- (NSInteger)tableView:(UITableView *)tableView numberOfRowsInSection:(NSInteger)section {
+- (void)scrollToCursorForTextView:(UITextView *)textView {
```

Comment:
The scrolling logic should go away at some point, because the cell will expand to contain the entire textfield and scrolling won't be necessary anymore.